File: src/app/models/offre-stage.ts
export interface Rapport {
  id: number;
  titre: string;
  document: string;
  nbCommentaires?: number;
  encadrant?: {
    id: number;
    nom: string;
    prenom: string;
    email: string;
    telephone?: string;
    poste?: string;
  };
}
export interface Tag {
  libelle: string;
  color: string;
}

export interface OffreStage {
  id: number;
  titre: string;
  description: string;
  date_debut: string; // LocalDate du backend
  date_fin: string;   // LocalDate du backend
  duree: number;      // Integer du backend
  statut: string;     // en_cours, fermee, archivee
  localisation: string;
  competence_requise: string;
  date_publication: string; // LocalDate du backend
  duree_candidature: number; // en jours
  nombre_limite_candidature?: number; // nullable - si null, pas de limite
  nombre_candidatures_actuelles?: number; // pour afficher le nombre actuel de candidatures
  entreprise?: {
    id: number;
    nom: string;
    logo?: string;
  };
}

File: src/main/java/com/gestionstage/gestionstage/entities/OffreStage.java
package com.gestionstage.gestionstage.entities;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;

@Entity
@Getter
@Setter
public class OffreStage {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

@ManyToOne(optional = false) 
@JoinColumn(name = "id_rh", nullable = false)
private Utilisateur rh;

    private String titre;
    private String description;
    private String sujet;

    public String getSujet() {
        return sujet;
    }

    private LocalDate date_debut;
    private LocalDate date_fin;
    private Integer duree;

    @Enumerated(EnumType.STRING)
    private StatutOffre statut = StatutOffre.en_cours;

    private String localisation;
    private String competence_requise;

    private LocalDate date_publication;
    private Integer duree_candidature; // en jours, peut être null
    private Integer nombre_limite_candidature; // peut être null

    public enum StatutOffre {
        en_cours, fermee, archivee
    }
}
